libs/designsystem/src/lib/components/button/button.component.scss

```diff
@@ -119,18 +87,28 @@ $button-width: (
   --kirby-button-color: #{utils.get-color('black')};
   --kirby-button-border-color: #{utils.get-color('medium')};
 
-  @include _button-hover(transparent, 'black', utils.get-loudness('faint'));
+  @include interaction-state.apply-hover('faint');
 
   &.destructive {
     --kirby-button-color: #{utils.get-color('danger')};
-
-    @include _button-hover(transparent, 'danger');
   }
 }
 
 :host {
   @include utils.accessible-target-size();
 
+  @include interaction-state.setup-layer;
+  @include interaction-state.apply-hover;
+  @include interaction-state.extend-content {
+    display: inherit;
+    flex-direction: inherit;
+    align-items: inherit;
+    justify-content: inherit;
+    padding-inline: var(--kirby-button-padding-left) var(--kirby-button-padding-right);
+  }
```

Comment:
What's the point of `extend-content`? Why not just write: 
```sass
.content {
    display: inherit;
    flex-direction: inherit;
    align-items: inherit;
    justify-content: inherit;
    padding-inline: var(--kirby-button-padding-left) var(--kirby-button-padding-right);
}
```

Reply:
It should not be applied to `.content` for all interaction state layers. And the `.content` rule block should not be applied directly in a component - hence the abstraction.